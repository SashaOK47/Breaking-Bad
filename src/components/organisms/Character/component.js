import PropTypes from "prop-types";
import style from "./styles.module.scss";
import Loader from "../../atoms/Loader";
import React from "react";

const Character = ({ character, isLoader }) => {
  const { img, name, status, nickname, birthday } = character;
  return isLoader ? (
    <Loader />
  ) : (
    <div>
      <div className={style.character}>
        <div className={style.photo}>
          <img className={style.img} src={img} alt={name} />
        </div>
        <div className={style.description}>
          <p className={style.status}>{status}</p>
          <p className={style.name}>{name}</p>
          <time className={style.date}>
            <span>Дата рождения:</span> {birthday}
          </time>
          <p className={style.nickname}>
            <span>Ник-нейм:</span> {nickname}
          </p>
          <blockquote className={style.quote}>
            <span>Цитата:</span> Lorem ipsum dolor sit amet, consectetur
            adipiscing elit. Non, aliquam laoreet vivamus sed. Diam fames mi,
            quam tellus cursus volutpat velit massa.
          </blockquote>
        </div>
      </div>
    </div>
  );
};

Character.propTypes = {
  character: PropTypes.object,
  isLoader: PropTypes.bool,
};
export default Character;
