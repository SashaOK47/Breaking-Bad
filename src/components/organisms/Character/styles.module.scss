@import "../../../assets/styles/main.scss";
.character {
  display: flex;
  @include tabletLg {
    flex-wrap: wrap;
    justify-content: center;
  }
  .photo {
    width: 36.5rem;
    max-height: 36.5rem;
    margin-right: 6.5rem;
    @include tabletLg {
      margin-right: 0;
      margin-bottom: 2rem;
    }
    .img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      object-position: top;
    }
  }
  .description {
    max-width: 25rem;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    @include tabletLg {
      align-items: center;
      text-align: center;
    }
    .status {
      padding: 0.5rem 0.75rem;
      margin-bottom: 2rem;
      background-color: $colorBgStatus;
      @include fontSmallCaps;
      @include tabletLg {
        margin-bottom: 1rem;
      }
    }
    .name {
      margin-bottom: 1rem;
      @include fontBig;
    }
    .date {
      margin-bottom: 1rem;
      span {
        color: $gray;
      }
    }
    .nickname {
      margin-bottom: 2rem;
      span {
        color: $gray;
      }
      @include tabletLg {
        margin-bottom: 1rem;
      }
    }
    .quote {
      width: 100%;
      position: relative;
      span {
        color: $gray;
        span {
          color: $mainFontColor;
        }
      }
    }
  }
}
