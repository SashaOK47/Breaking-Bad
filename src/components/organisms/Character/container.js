import React from "react";
import PropTypes from "prop-types";
import Character from "./component";
import Error from "../../atoms/Error";
import { useSelector } from "react-redux";

const CharacterContainer = ({ character, isLoader }) => {
  const isError = useSelector((state) => state.characters.isError);

  return isError ? (
    <Error message={isError} />
  ) : (
    <Character character={character} isLoader={isLoader} />
  );
};

CharacterContainer.propTypes = {
  character: PropTypes.object,
  isLoader: PropTypes.bool,
};
export const container = CharacterContainer;
