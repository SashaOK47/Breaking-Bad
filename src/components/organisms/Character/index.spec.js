import React from "react";
import Character from "./component";

const character = {
  char_id: 1,
  name: "Test",
  birthday: "12-12-12",
  img: "https://test.test",
  status: "Test status",
  nickname: "Test Nickname",
};
const quote = "Test quote";
let isLoader = true;
let isLoaderQuote = true;

const setUp = (props) => shallow(<Character {...props} />);

describe("Character component", () => {
  it("Character render isLoader", () => {
    const component = setUp({ isLoader, character });
    expect(component).toMatchSnapshot();
  });
  it("Character render after isLoader", () => {
    isLoader = false;
    const component = setUp({ isLoader, character });
    expect(component).toMatchSnapshot();
  });
  it("should render isLoaderQuote", () => {
    isLoader = false;
    const component = setUp({ isLoader, character, isLoaderQuote, quote });
    expect(component).toMatchSnapshot();
  });
  it("Quote render after isLoaderQuote", () => {
    isLoader = false;
    isLoaderQuote = false;
    const component = setUp({ isLoader, character, isLoaderQuote, quote });
    expect(component).toMatchSnapshot();
  });
});
