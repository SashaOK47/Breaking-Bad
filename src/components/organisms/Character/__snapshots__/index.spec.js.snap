// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component Character render after isLoader 1`] = `
<div>
  <div
    className="character"
  >
    <div
      className="photo"
    >
      <img
        alt="Test"
        className="img"
        src="https://test.test"
      />
    </div>
    <div
      className="description"
    >
      <p
        className="status"
      >
        Test status
      </p>
      <p
        className="name"
      >
        Test
      </p>
      <time
        className="date"
      >
        <span>
          Дата рождения:
        </span>
         
        12-12-12
      </time>
      <p
        className="nickname"
      >
        <span>
          Ник-нейм:
        </span>
         
        Test Nickname
      </p>
      <blockquote
        className="quote"
      >
        <span>
          Цитата: 
        </span>
        <span>
           
        </span>
      </blockquote>
    </div>
  </div>
</div>
`;

exports[`Character component Character render isLoader 1`] = `<Loader />`;

exports[`Character component Quote render after isLoaderQuote 1`] = `
<div>
  <div
    className="character"
  >
    <div
      className="photo"
    >
      <img
        alt="Test"
        className="img"
        src="https://test.test"
      />
    </div>
    <div
      className="description"
    >
      <p
        className="status"
      >
        Test status
      </p>
      <p
        className="name"
      >
        Test
      </p>
      <time
        className="date"
      >
        <span>
          Дата рождения:
        </span>
         
        12-12-12
      </time>
      <p
        className="nickname"
      >
        <span>
          Ник-нейм:
        </span>
         
        Test Nickname
      </p>
      <blockquote
        className="quote"
      >
        <span>
          Цитата: 
        </span>
        <span>
           
          Test quote
        </span>
      </blockquote>
    </div>
  </div>
</div>
`;

exports[`Character component should render isLoaderQuote 1`] = `
<div>
  <div
    className="character"
  >
    <div
      className="photo"
    >
      <img
        alt="Test"
        className="img"
        src="https://test.test"
      />
    </div>
    <div
      className="description"
    >
      <p
        className="status"
      >
        Test status
      </p>
      <p
        className="name"
      >
        Test
      </p>
      <time
        className="date"
      >
        <span>
          Дата рождения:
        </span>
         
        12-12-12
      </time>
      <p
        className="nickname"
      >
        <span>
          Ник-нейм:
        </span>
         
        Test Nickname
      </p>
      <blockquote
        className="quote"
      >
        <span>
          Цитата: 
        </span>
        <Loader
          small={true}
        />
      </blockquote>
    </div>
  </div>
</div>
`;
